src/similarity.spec.ts

```diff
@@ -0,0 +1,333 @@
+          "user-agent": "Chrome",
+          host: "local",
+          connection: "persist",
+          "x-region": "australia"
+        },
+        Buffer.from([]),
+        {
+          request: {
+            method: "POST",
+            path: "/test",
+            headers: {
+              accept: "application/vnd.api+json",
+              "user-agent": "Firefox",
+              host: "google.com",
+              "x-region": "UK"
+            },
+            body: Buffer.from([])
+          },
+          response: DUMMY_RESPONSE
+        }
+      )
+    ).toBe(1);
+  });
+
+  it("relies on JSON payload similarity", () => {
+    // The following payloads are identical, but formatted differently.
+    expect(
+      computeSimilarity(
+        "POST",
+        "/test",
+        {},
+        compactJsonBuffer({
+          a: 1,
+          b: 2,
+          c: 3
+        }),
+        {
+          request: {
+            method: "POST",
+            path: "/test",
+            headers: {},
+            body: wellFormattedJsonBuffer({
+              a: 1,
+              b: 2,
+              c: 3
+            })
+          },
+          response: DUMMY_RESPONSE
+        }
+      )
+    ).toBe(0);
+
+    // The following payloads only have one field that differs (c).
+    expect(
+      computeSimilarity(
+        "POST",
+        "/test",
+        {},
+        compactJsonBuffer({
+          a: 1,
+          b: 2,
+          c: 3
+        }),
+        {
+          request: {
+            method: "POST",
+            path: "/test",
+            headers: {},
+            body: wellFormattedJsonBuffer({
+              a: 1,
+              b: 2,
+              c: {
+                d: 4,
+                e: 5
+              }
+            })
+          },
+          response: DUMMY_RESPONSE
+        }
+      )
+    ).toBe(1);
+
+    // The following payloads have all different fields.
+    expect(
+      computeSimilarity(
+        "POST",
+        "/test",
+        {},
+        compactJsonBuffer({
+          a: 1,
+          b: 2,
+          c: 3
+        }),
+        {
+          request: {
+            method: "POST",
+            path: "/test",
+            headers: {},
+            body: wellFormattedJsonBuffer({
+              d: 1,
+              e: 2,
+              f: 3
+            })
+          },
+          response: DUMMY_RESPONSE
+        }
+      )
+    ).toBe(6);
+  });
+
+  it("relies on string similarity", () => {
+    expect(
+      computeSimilarity("POST", "/test", {}, Buffer.from("abc"), {
+        request: {
+          method: "POST",
+          path: "/test",
+          headers: {},
+          body: Buffer.from("abc")
+        },
+        response: DUMMY_RESPONSE
+      })
+    ).toBe(0);
+    expect(
+      computeSimilarity("POST", "/test", {}, Buffer.from("hello world"), {
+        request: {
+          method: "POST",
+          path: "/test",
+          headers: {},
+          body: Buffer.from("hello Kevin")
```

Comment:
Is this a test to see if I read all the test code? 🤔

Reply:
I think Hamming distance might be better for string comparison - no need to do it now tho.

E.g., `hello 1` and `hello 2` should yield a better difference than `hello 1` and `abcdefg`. If I read the code correctly, currently the difference is the same.

Reply:
I'll dig into this deeper separately, yeah.